uri module: added use_netrc argument to allow disabling netrc (#74397)

Artur-at-work opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Added `use_netrc` argument to control whether to use or ignore the .netrc (default: True) 
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
Fixes #74397
##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
uri
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
After changes verified below behavior using "python3 AnsiballZ_uri.py execute"
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Lillian/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Lillian/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
.font5
	{color:windowtext;
	font-size:9.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:新細明體, serif;
	mso-font-charset:136;}
.font6
	{color:black;
	font-size:8.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:"DejaVu Sans Mono", monospace;
	mso-font-charset:0;}
.font7
	{color:#4E9A06;
	font-size:8.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:"DejaVu Sans Mono", monospace;
	mso-font-charset:0;}
.font8
	{color:#AD7FA8;
	font-size:8.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:"DejaVu Sans Mono", monospace;
	mso-font-charset:0;}
tr
	{mso-height-source:auto;
	mso-ruby-visibility:none;}
col
	{mso-width-source:auto;
	mso-ruby-visibility:none;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:新細明體, serif;
	mso-font-charset:136;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{border:.5pt solid windowtext;}
.xl66
	{color:#204A87;
	font-size:8.0pt;
	font-family:"DejaVu Sans Mono", monospace;
	mso-font-charset:0;
	text-align:left;
	vertical-align:middle;
	border:.5pt solid windowtext;
	white-space:normal;
	padding-left:12px;
	mso-char-indent-count:1;}
ruby
	{ruby-align:left;}
rt
	{color:windowtext;
	font-size:9.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:新細明體, serif;
	mso-font-charset:136;
	mso-char-type:none;
	display:none;}
-->
</head>

<body link="#0563C1" vlink="#954F72">



  | Before Fix   (use_netrc=True) | After Fix   (use_netrc=False)
-- | -- | --
"url":"https://httpbin.org/bearer", | "status":401, | "status":200,
"url":"https://httpbin.org/anything", | "Authorization":"Basic Zm9vOmJhcg==", | "Authorization":"Bearer foobar",



</body>

</html>

- Sanity Tests - PASSED
- Integration Test- PASSED for uri module. I've got some "ignored" cases, but no "failed"
- Unit Tests - some existing cases failed because this fix added one more argument. Updated those cases by adding "use_netrc=True" attribute to the asserts. Then PASSED

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

mattclay committed:
Add missing space after keywords. (#78480)

Also remove unnecessary parenthesis.

akira6592 committed:
fix configuration name in porting guide 2.12  (#78472)

akira6592 committed:
Python 3.9 Controller minimum (#78466)

Alex-Izquierdo committed:
update the documentation for the _collection init_ command (#78404)

mdonoughe committed:
Update attributes of files that are links without specifying link target (#76167)

* update attributes of files that are symlinks

* update attributes of files that are hard links

* fix default state in documentation

* remove unnecessary suppression

* add to changelog

mattclay committed:
ansible-test - Sanity test code cleanup. (#78497)

mattclay committed:
Clean up unit tests. (#78498)

mattclay committed:
ansible-test - Update coverage to 6.4.3. (#78504)

mattclay committed:
ansible-test - No locale warning with delegation. (#78505)

The locale warning for the origin host is not relevant when delegation is used.
It is also not relevant when non-test commands are in use.

mattclay committed:
ansible-test - Update distro test containers. (#78506)

Artur-at-work committed:
uri module: added use_netrc argument to allow disabling netrc (#74397)

Artur-at-work committed:
uri: added use_netrc argument to allow disabling netrc (#74397)

samccann committed:
remove old network link (#78495)

mattclay committed:
module_utils - Fix changing value while iterating. (#78501)

mattclay committed:
Remove unused code. (#78499)

mattclay committed:
find - Use `getattr` instead of `__getattribute__` (#78503)

mattclay committed:
ansible-test - Fix ansible-doc sanity test FQCN. (#78518)

akira6592 committed:
docs: misc fixes in The Ansible Development Cycle (#78513)

mattclay committed:
ansible-console - Fix arg name to match base class. (#78502)

s-hertel committed:
Fix apt making changes in check mode (#78496)

* Don't actually update the cache in check mode

Add tests for updating the cache in check mode

* Don't mark packages as manually installed in check mode and add a test for it

* changelog

* Update test/integration/targets/apt/tasks/apt.yml

* fix test

Remi-Mergen committed:
adding .PHONY for 3 rules (#78510)

rules: docs, version, install_manpages

bcoca commented on lib/ansible/modules/uri.py:
```diff
@@ -205,6 +205,14 @@
+      - When set to False, .netrc credentials are ignored
+    type: bool
+    default: true
+    version_added: '2.13'
```
2.14 at this point

bcoca reviewed (changes requested):
A changelog and integration tests would be required for merge

mkrizek committed:
PlayIterator: deprecate cache_block_tasks and get_original_task (#78514)

bcoca committed:
ansible-config init fixes (#78453)

* ansible-config init fixes

  now handles --disabled correctly for 'vars' format
  also does not display internal config entries anymore

  fixes #78438

note plugins have _prefixed settings that are public, this should change

sivel committed:
Fix --role-file arg detection (#78475)

* Fix --role-file arg detection. Fixes #78204

* Do not traceback, give better error

* Add coverage for compound shortopts to match -r. Fixes #78491

mattclay committed:
ansible-test - Fix self-test change handling. (#78521)

bcoca committed:
add info about plugin name when ignoring abstract errors (#78465)

* add info about plugin name when ignoring abstract

  related to #78464

* added error to narrow down method

bcoca committed:
allow multiple types for null representation (#78451)

* allow multiple types for null representation

  '' is not Nonetype
  fixes #76493

mattclay committed:
Split up ansible-test integration tests. (#78520)

* Relocate venv-pythons.py script.

* Split out unsupported-directory test.

* Split out sanity-import test.

* Split out sanity-validate-modules test.

* Split out units test.

* Split out integration test.

* Split out units-constraints test.

* Split out integration-constraints test.

* Split out coverage test.

* Split out sanity test.

* Split out git test.

* Update test groups.

akamanzi committed:
Fix ansible-docs: shipping ansible in wheel format  #78485 (#78524)

mattclay committed:
Fix variable use before definition. (#78500)

* Fix variable use before definition.

* Include mkstemp in exception handler.

Also remove two pointless variable assignments.

mattclay committed:
ansible-test - Update sanity test requirements. (#78528)

mattclay committed:
ansible-test - Update base and default containers. (#78532)

Artur-at-work commented:
Sorry for the mess with other commits, I only meant to add my own commit:
https://github.com/ansible/ansible/tree/d191102c1df94bc6edcd1b943ab8b5a90d384250

It seems I can't remove those manually. 
Is it cleaner if I abandon/remove this PR and submit a new one? 

Please let me know if there a better way.
